spokestack/profile/vad_trigger_asr.py

```diff
@@ -0,0 +1,59 @@
+"""
+Pipeline profile with vad trigger and asr
+"""
+from spokestack.activation_timeout import ActivationTimeout
+from spokestack.asr.speech_recognizer import CloudSpeechRecognizer
+from spokestack.io.pyaudio import PyAudioInput
+from spokestack.pipeline import SpeechPipeline
+from spokestack.vad.webrtc import VoiceActivityDetector, VoiceActivityTrigger
+
+
+class VoiceActivityTriggerASR:
+    """ VAD Trigger ASR
+
+    Args:
+            spokestack_id (str):
+            spokestack_secret (str):
+            sample_rate (int):
+            frame_width (int):
+            vad_rise_delay (int):
+            vad_fall_delay (int):
+            idle_timeout (int):
+            mode (int):
+
+    """
+
+    def __init__(
+        self,
+        spokestack_id: str,
+        spokestack_secret: str,
+        sample_rate: int = 16000,
+        frame_width: int = 20,
+        **kwargs
+    ) -> None:
+        self._pipeline = SpeechPipeline(
+            input_source=PyAudioInput(
+                sample_rate=sample_rate, frame_width=frame_width, **kwargs
+            ),
+            stages=[
+                VoiceActivityDetector(
+                    sample_rate=sample_rate, frame_width=frame_width, **kwargs
+                ),
+                VoiceActivityTrigger(),
+                ActivationTimeout(frame_width=frame_width, **kwargs),
+                CloudSpeechRecognizer(
+                    spokestack_id=spokestack_id,
+                    spokestack_secret=spokestack_secret,
+                    sample_rate=sample_rate,
+                    frame_width=frame_width,
+                    **kwargs
+                ),
+            ],
+        )
+
+    def run(self) -> None:
```

Comment:
the problem with using the profile to hide the pipeline is that you lose all the other features of the pipeline (events, etc.)
it would be better to have the profile construct and return the pipeline, using a `create` method or whatever

Reply:
one thought I had originally was being able to add the event handlers like `@profile.pipeline.event`

Reply:
I think that the primary thing the application interacts with is the pipeline, whereas the profile just provides some boilerplate configuration and wiring